source/common/network/io_socket_handle_impl.cc

```diff
@@ -56,9 +56,16 @@ IoSocketHandleImpl::~IoSocketHandleImpl() {
   if (SOCKET_VALID(fd_)) {
     IoSocketHandleImpl::close();
   }
+  if (file_event_) {
+    file_event_.reset();
+  }
```

Comment:
Ok that makes sense, I was just trying hard not to leak. Although, it should be needed anyway since it is a smart pointer and it is life time is managed.